apps/scan/frontend/src/app_root.tsx

```diff
@@ -176,7 +182,13 @@ export function AppRoot({ hardware, logger }: Props): JSX.Element | null {
     return (
       <UnlockMachineScreen
         auth={authStatus}
-        checkPin={(pin: string) => checkPinMutation.mutate({ pin })}
+        checkPin={async (pin) => {
+          try {
+            await checkPinMutation.mutateAsync({ pin });
+          } catch {
+            // Handled by default query client error handling
```

Comment:
I'm confused by how the default query client error handling works. It would seem to me that here you're catching and _suppressing_ the error from hitting the error boundary, which is the "default query client error handling" I think you're talking about?

Reply:
Yeah it's definitely not the most intuitive pattern 😅

The default query client error handling is happening inside the `.mutateAsync` call. But even despite that handling, the `.mutateAsync` will still reject. The try-catch prevents an unhandled promise rejection. I actually pulled this pattern straight from one of @jonahkagan's slides from his re-architecture tour ([relevant slide](https://docs.google.com/presentation/d/1qcohhkeavCVjTZAAhGbbB7sbnvDJLlNov3OTLijfcvs/edit#slide=id.g1f8414d696e_0_361))

Reply:
Weird that the react error bubbling and the query client error bubbling have different paths, but I suppose that's just how it works. Sounds good.